File: lib/features/auth/data/models/user_dto.dart
class UserDto {
  final String id; // Guid from backend
  final String username;
  final String email;
  final List<String> permissions;
  final String token;

  UserDto({
    required this.id,
    required this.username,
    required this.email,
    required this.permissions,
    required this.token,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      id: json['id'] ?? '',
      username: json['username'] ?? '',
      email: json['email'] ?? '',
      permissions: List<String>.from(json['permissions'] ?? []),
      token: json['token'] ?? '',
    );
  }
}

File: backend/src/EnterpriseAuth.Api/Core/Application/DTOs/UserDto.cs
using System;

namespace EnterpriseAuth.Api.Core.Application.DTOs
{
    public class UserDto
    {
        public Guid Id { get; set; }
        public string Email { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public bool IsActive { get; set; }
        public Guid? UserGroupId { get; set; }
        public List<string> Permissions { get; set; } = new List<string>();
    }

    public class UserCreateRequest
    {
        public string Email { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public Guid UserGroupId { get; set; }
        public List<string> Permissions { get; set; } = new List<string>();
    }
}
